rotkehlchen/chain/ethereum/modules/convex/decoder.py

```diff
@@ -192,7 +200,41 @@ def _decode_gauge_events(self, context: DecoderContext) -> DecodingOutput:
                         event.notes = f'Claim {event.balance.amount} {crypto_asset.symbol} reward from convex {self.pools[context.tx_log.address]} pool'  # noqa: E501
                     else:
                         event.notes = f'Claim {event.balance.amount} {crypto_asset.symbol} reward from convex'  # noqa: E501
-        return DecodingOutput(refresh_balances=found_event_modifying_balances)
+        return DecodingOutput(
+            refresh_balances=found_event_modifying_balances,
+            matched_counterparty=matched_counterparty,
+        )
+
+    def _check_lock_withdrawal_funds(
+            self,
+            transaction: 'EvmTransaction',  # pylint: disable=unused-argument
+            decoded_events: list['EvmEvent'],
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        withdrawals_log_entries = filter(lambda x: x.topics[0] == CVX_LOCK_WITHDRAWN, all_logs)
+        amounts_withdrawn = [
+            asset_normalized_value(hex_or_bytes_to_int(tx_log.data[0:32]), self.cvx)
+            for tx_log in withdrawals_log_entries
+        ]  # needs to be a list since someone could lock in different contracts the same amount
+
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == A_CVX and
+                event.balance.amount in amounts_withdrawn
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_CONVEX
+                event.notes = f'Unlock {event.balance.amount} {self.cvx.symbol} from convex'
+
+                # stop as soon as we have processed all the unlock events
+                amounts_withdrawn.remove(event.balance.amount)
+                if len(amounts_withdrawn) == 0:
+                    break
```

Comment:
what happens if not all are processed? Add a log entry after the loop

Reply:
If all the events for some reason don't get processed we iterate over all the decoded events in the worst case. I designed it like this in the case of someone calling it in a multicall or something

Reply:
That's not what I mean. If at the very end `len(amounts_withdrawn) != 0` what happens? Chat with me in discord about it.